analyses/tmb-vaf-longitudinal/03-explore-TMB-mut.Rmd

```diff
@@ -0,0 +1,221 @@
+cg_sum_df <- df %>% 
+  count(cg_sum) %>% 
+  dplyr::mutate(cg_sum_n = glue::glue("{cg_sum} (N={n})"))
+
+df <- df %>% 
+  left_join(cg_sum_df, by = c("cg_sum")) %>% 
+  mutate(cg_plot = case_when(n < 10 ~ "Other",
+                             TRUE ~ cg_sum))
+
+# How many bs_samples per cg_plot?
+print(df %>% count(cg_plot) %>% arrange(desc(n))) 
+
+# Read color palette
+palette_df <- readr::read_tsv(palette_file, guess_max = 100000, show_col_types = FALSE)
+
+# length(unique(df$Kids_First_Participant_ID))
+```
+
+# TMB per Patient case
+We will explore TMB per `Kids_First_Participant_ID` over time by creating stacked barplots.
+
+```{r create-stacked-barplot, fig.width = 15, fig.height = 6, fig.fullwidth = TRUE}
+# Define parameters for function
+ylim = max(df$tmb)
+
+# Re-order df
+f <- c("Second Malignancy", "Unavailable", "Deceased", "Recurrence", "Progressive", "Diagnosis") # Level df by timepoints
+df_plot <- df %>% 
+  dplyr:::mutate(tumor_descriptor = factor(tumor_descriptor),
+                 tumor_descriptor = fct_relevel(tumor_descriptor, f)) 
+
+# Run function
+fname <- paste0(plots_dir, "/", "TMB-genomic-total.pdf")
+print(fname)
+p <- create_stacked_barplot(tmb_df = df_plot, ylim = ylim)
+pdf(file = fname, width = 15, height = 6)
+print(p)
+dev.off()
+```
+Attention: Hypermutant TMB defined as ≥10 Mb, and Ultrahypermutant TMB defined as ≥100 mutations/Mb in pediatric brain tumors (https://pubmed.ncbi.nlm.nih.gov/29056344/).
+
+Here, we notice that there are samples with high TMB (hyper-mutant samples). Next, we will exclude these samples (threshold >= 10) from downstream analysis. Attention is needed in cases with high number of mutations in only one timepoint as this will lead to un-matched longitudinal samples. We will also remove those so we always have matched longitudinal samples.
+
+```{r create-stacked-barplot-filter, fig.width = 15, fig.height = 8, fig.fullwidth = TRUE}
+# Filter df and remove any samples with single timepoints
+df_plot_filter <- df %>%
+  filter(!tmb >= 10) %>%
+  unique() %>% 
+  arrange(Kids_First_Participant_ID, tumor_descriptor) %>%
+  group_by(Kids_First_Participant_ID) %>%
+  dplyr::summarise(tumor_descriptor_sum = str_c(tumor_descriptor, collapse = ";")) %>% 
+  filter(!tumor_descriptor_sum %in% c("Diagnosis", "Progressive", "Recurrence", "Second Malignancy", "Unavailable", "Deceased", "Progressive;Progressive")) %>% 
+  dplyr::left_join(df, by = c("Kids_First_Participant_ID", "tumor_descriptor_sum")) %>% 
+  mutate(tumor_descriptor = factor(tumor_descriptor),
+         tumor_descriptor = fct_relevel(tumor_descriptor, f)) %>% 
+  drop_na(tmb) 
+
+# length(unique(df_plot_filter$Kids_First_Participant_ID))
+
+# Define parameters for function
+ylim = max(df_plot_filter$tmb)
+df_plot_filter <- df_plot_filter
+
+# Run function
+fname <- paste0(plots_dir, "/", "TMB-genomic-no-hypermutants.pdf")
+print(fname)
+p <- create_stacked_barplot(tmb_df = df_plot_filter, ylim = ylim)
```

Comment:
This looks good! Is there any way you could group patients by cancer group here?

Reply:
Done, see 504c0590334c6c8520e5698f01ec50f83711e4af